### PR TITLE
1994 - adds pip install tests

### LINKED ISSUE #1994: Set up CI/CD for JIT extension packaging/release
**Is your feature request related to a problem? Please describe.**
follow up of PR #1802, CI/CD should be updated to support the packaging and release of the new features https://github.com/Project-MONAI/MONAI/pull/1802#issuecomment-814838242

### PR DESCRIPTION
**Pull request by wyli**
Fixes #1994

- adds package min_tests for pre-merge ci/cd
- adds package full unit tests for nightly ci/cd

### Status
**Ready**

tested:
- https://github.com/Project-MONAI/MONAI/runs/2647583616
- https://github.com/Project-MONAI/MONAI/runs/2647355573?check_suite_focus=true

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.

### CODE REVIEW

File: .github/workflows/cron.yml [hunk 1]
```diff
@@ -99,6 +99,73 @@ jobs:
...
+
+          # move packages to a temp dir
+          tmp_dir=$(mktemp -d)
+          cp dist/monai* "$tmp_dir"
+          rm -r build dist monai.egg-info
+          cd "$tmp_dir"
+          ls -al
+
+          # install from tar.gz
+          name=$(ls *.tar.gz | head -n1)
```
Comment (rijobro) [inline .github/workflows/cron.yml:144]: Sorry, where does the tar.gz come from?

**Comment by rijobro** [comment]
Am I correct in understanding that this PR doesn't install from pypi, but from tar.gz? Do you think this changes the "realness" of the test at all?

File: .github/workflows/cron.yml (hunk 1 above)
Comment (wyli) [reply .github/workflows/cron.yml:144]: sure,created at line133, copied at line 138

**Comment by wyli** [comment]
> Am I correct in understanding that this PR doesn't install from pypi, but from tar.gz? Do you think this changes the "realness" of the test at all?

yes this is quite realistic, except that "pip install requirements-dev.txt" might hide some dependency errors. but we need it to construct a testing environment. we should consider refactoring requirements*.txt.

for the tar.gz, another approach is to upload the tar.gz to testPyPI server and pip install from there.  this pipeline will then depend on the availability of the test PyPI server, and for now I don't think it's really needed.

**Comment by rijobro** [comment]
Ok, it sounds good to me. I can't really offer an opinion as to best practice, but happy to approve.

Has the workflow been run? Looking at the cron jobs and can't see `cron-pip` anywhere.

image

**Comment by wyli** [comment]
thanks, the cron will be availabe after merging the config. I had a test separately here: https://github.com/Project-MONAI/MONAI/actions/runs/867438967

**Review by rijobro — APPROVED** [review]
Great, thanks!

**Pull request merged**